src/plugins/intel_gna/src/transformations/transpose_compress.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/transpose_compress.hpp"
+
+#include "common/graph_utils.hpp"
+#include "openvino/cc/ngraph/itt.hpp"
+#include "openvino/opsets/opset11.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+using namespace ov::opset11;
+using namespace ov::pass;
+using namespace ov::intel_gna::pass;
+
+namespace {
+
+inline std::vector<size_t> fix_indexes(const std::vector<size_t>& ids) {
+    std::vector<size_t> ids_fixed(ids.size());
+    std::iota(ids_fixed.begin(), ids_fixed.end(), 0);
+    stable_sort(ids_fixed.begin(), ids_fixed.end(), [&ids](size_t a, size_t b) {
+        return ids[a] < ids[b];
+    });
+    return ids_fixed;
+}
+
+}  // namespace
+
+TransposeCompress::TransposeCompress() {
+    MATCHER_SCOPE(TransposeCompress);
+
+    auto transpose_const = pattern::wrap_type<Constant>();
+    auto transpose =
+        pattern::wrap_type<Transpose>({pattern::any_input(), transpose_const}, [](const ov::Output<ov::Node>& node) {
+            return !limitations::Limitations::is_transpose_supported(node.get_node_shared_ptr());
```

Comment:
is_one_dim_shape can be used combined here.

Reply:
discussed offline to leave the current approach